src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Utilities/EnsureOnceInitializedOnceDisposedAsync.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem
+{
+    internal abstract class EnsureOnceInitializedOnceDisposedAsync : OnceInitializedOnceDisposedAsync
```

Comment:
This isn't needed, OnceInitializedOnceDisposedAsync already gives this guarantee.